papers/DX_zstring_view.fiets

```diff
@@ -81,6 +142,211 @@ The `substr` function is changed from a single function with a default argument,
+      constexpr u8zstring_view  operator""zsv(const char8_t* str, size_t len) noexcept;
+      constexpr u16zstring_view operator""zsv(const char16_t* str, size_t len) noexcept;
+      constexpr u32zstring_view operator""zsv(const char32_t* str, size_t len) noexcept;
+      constexpr wzstring_view   operator""zsv(const wchar_t* str, size_t len) noexcept;
+    }
+  }
+}
+```
+
+### Class template basic_zstring_view [zstring.view.template]
+
+### General [zstring.view.template.general]
+
+```cpp
+namespace std {
+  template<class charT, class traits = char_traits<charT>>
+  class basic_zstring_view {
+  public:
+    // types
+    using traits_type            = traits;
+    using value_type             = charT;
+    using pointer                = value_type*;
+    using const_pointer          = const value_type*;
+    using reference              = value_type&;
+    using const_reference        = const value_type&;
+    using const_iterator         = implementation-defined; // see [zstring.view.iterators]
+    using iterator               = const_iterator;
+    using const_reverse_iterator = reverse_iterator<const_iterator>;
+    using reverse_iterator       = const_reverse_iterator;
+    using size_type              = size_t;
+    using difference_type        = ptrdiff_t;
+    static constexpr size_type npos = size_type(-1);
+
+    // [zstring.view.cons], construction and assignment
+    constexpr basic_zstring_view() noexcept;
+    constexpr basic_zstring_view(const basic_zstring_view&) noexcept = default;
+    constexpr basic_zstring_view& operator=(const basic_zstring_view&) noexcept = default;
+    constexpr basic_zstring_view(const charT* str);
+    constexpr basic_string_view(const charT* str, size_type len);
+    basic_zstring_view(nullptr_t) = delete;
+
+    // [zstring.view.iterators], iterator support
+    constexpr const_iterator begin() const noexcept;
+    constexpr const_iterator end() const noexcept;
+    constexpr const_iterator cbegin() const noexcept;
+    constexpr const_iterator cend() const noexcept;
+    constexpr const_reverse_iterator rbegin() const noexcept;
+    constexpr const_reverse_iterator rend() const noexcept;
+    constexpr const_reverse_iterator crbegin() const noexcept;
+    constexpr const_reverse_iterator crend() const noexcept;
+
+    // [zstring.view.capacity], capacity
+    constexpr size_type size() const noexcept;
+    constexpr size_type length() const noexcept;
+    constexpr size_type max_size() const noexcept;
+    [[nodiscard]] constexpr bool empty() const noexcept;
+
+    // [zstring.view.access], element access
+    constexpr const_reference operator[](size_type pos) const;
+    constexpr const_reference at(size_type pos) const;
+    constexpr const_reference front() const;
+    constexpr const_reference back() const;
+    constexpr const_pointer data() const noexcept;
+    constexpr const_pointer c_str() const noexcept;
+
+    operator basic_string_view<charT, traits>() const noexcept;
```

Comment:
I think we want this conversion operator (and if I'm reading "zstring_view and string_view relation" correctly that seems to be consistent). I've made some of the later zstring_view operations take a string_view as a result.